### PR TITLE
Handle tasks asynchronously

### PR DESCRIPTION
**Pull request by Threated**
Review commit by commit.

I will test this branch on the DNPM server before merging

### CODE REVIEW

**Comment by Threated** [comment]
Running flawlessly on the central dnpm servers

File: src/logic_ask.rs [hunk 1]
```diff
@@ -147,19 +145,17 @@ async fn handle_via_tasks(req: Request<Incoming>, config: &Arc<Config>, target:
...
+        return Err(StatusCode::INTERNAL_SERVER_ERROR.into());
+    };
+    debug!("Got reply with status {:?}: {:#?}", result.status, result.body);
     let response_inner = match result.status {
         WorkStatus::Succeeded => {
             result.body
         },
         e => {
-            warn!("Reply had unexpected workresult code: {e:?}");
+            warn!("Reply had unexpected workresult code: {e:?}: {:#?}", result.body);
```
Comment (TKussel) [inline src/logic_ask.rs:158]: I'm a bit concerned about his. It only triggers at an error status code, right? So we don't run into the risk of logging sensitive data by printing the body?

File: src/main.rs [hunk 2]
```diff
@@ -28,7 +28,7 @@ async fn main() -> anyhow::Result<()> {
     tracing::subscriber::set_global_default(tracing_subscriber::fmt().with_env_filter(EnvFilter::builder().with_default_directive(LevelFilter::INFO.into()).from_env_lossy()).finish())?;
     banner::print_banner();
     let config = Config::load().await?;
-    let config2 = config.clone();
+    let config: &'static _ = Box::leak(Box::new(config));
```
Comment (TKussel) [inline src/main.rs:31]: Thanks for showing me this trick!

File: src/msg.rs [hunk 3]
```diff
@@ -23,6 +23,16 @@ pub(crate) struct HttpResponse {
     pub(crate) body: Vec<u8>
 }
 
+impl std::fmt::Debug for HttpResponse {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("HttpResponse")
+            .field("status", &self.status)
+            .field("headers", &self.headers)
+            .field("body", &String::from_utf8_lossy(&self.body).as_ref())
```
Comment (TKussel) [inline src/msg.rs:31]: Again, please check if we debug print the respones somewhere, where sensitive data might be in the body

**Review by TKussel — APPROVED** [review]

File: src/logic_ask.rs (hunk 1 above)
Comment (Threated) [reply src/logic_ask.rs:158]: Yes although I just realized we won't even get here as a few lines earlier we try to deserialize into a `Something<HttpResponse>` which won't work and error early so to improve the error messages I will change that to deserialize into `serde_json::Value`s first and if the status is success then into the `HttpResponse`. What do you think?

File: src/msg.rs (hunk 3 above)
Comment (Threated) [reply src/msg.rs:31]: Will double check good catch

Comment (Threated) [reply src/msg.rs:31]: Would logging headers still be fine?
We could also have it include the body if its build in debug mode and skip it in release builds

Comment (TKussel) [reply src/msg.rs]: As we are logging the *response*, there should be no e.g. `Authorization` header present. So I think that should be fine? 
Not logging the bodies in release builds is certainly the secure path forward.

**Pull request merged**